### ISSUE TITLE
Rule proposal - Readability rule - Report redundant return at the end of void method

### ISSUE THREAD

**Issue by Przemyslaw-W** [DotNetAnalyzers/StyleCopAnalyzers#760]
```
void Func() 
{ 
    ... 
    return; 
} 
```
The return is useless, so I think it should be flagged.

Originally proposed by Xanatos http://stylecop.codeplex.com/workitem/6801

------------------------------------------------------

follow up discussion:

andyr wrote Feb 9, 2011 at 3:54 PM
New rule will be added.

xanatos wrote Feb 10, 2011 at 9:05 AM
The same should (probably) be for
```
    public IEnumerable<int> Func()
    {
        .....
        yield break;
    }
```
(yield break at the end of a method)

hfrmobile wrote Feb 26, 2011 at 11:55 AM
Voted.

**Comment by Przemyslaw-W** [comment]
I like it

**Comment by sharwell** [comment]
I can't even remember the last time I saw this happen.

**Comment by Przemyslaw-W** [comment]
I never saw set before get. Yet, rule does exist

**Comment by vweijsters** [comment]
I've seen this a few times in the past, but not often. A friendly chat with the responsible developer solved it and made sure it never appeared again :smiling_imp: 

I think we can include it, but it wouldn't be high on my list of things to add.

**Comment by GregReddick** [comment]
Newby programmers and people coming from other languages might do this. I'm ambivalent about the rule. We have rules for many things that I would never do, but shouldn't be allowed.
